homeassistant/helpers/translation.py

```diff
@@ -323,3 +337,111 @@ async def async_get_translations(
+        else:
+            cache = hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
+        await cache.async_load(language, components)
+
+
+@callback
+def async_get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> dict[str, Any]:
+    """Return all cached backend translations.
+
+    If integrations specified, return translations for them.
+    Otherwise default to loaded integrations.
+    """
+    components = _async_get_components_to_load(hass, category, integrations)
+
+    if TRANSLATION_FLATTEN_CACHE not in hass.data:
+        return {}
+
+    result = {}
+
+    cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+    cached = cache.get_cached(language, category, components)
+    for entry in cached:
+        result.update(entry)
+
+    return result
+
+
+@callback
+def _async_get_components_to_load(
+    hass: HomeAssistant,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> set[str]:
+    if integrations is not None:
+        components = set(integrations)
+    elif category in ("state", "entity_component", "services"):
+        components = set(hass.config.components)
+    else:
+        # Only 'state' supports merging, so remove platforms from selection
+        components = {
+            component for component in hass.config.components if "." not in component
+        }
+    return components
+
+
+async def async_load_state_translations_to_cache(
+    hass: HomeAssistant,
+    language: str,
+    integrations: Iterable[str] | None = None,
+) -> None:
+    """Load state translations to cache."""
+    await async_load_translations(hass, language, "entity", integrations)
+    await async_load_translations(hass, language, "state", integrations)
+    await async_load_translations(hass, language, "entity_component", integrations)
+
+
+@callback
+def async_setup_load_listeners(hass: HomeAssistant) -> None:
+    """Register listeners for translation loaders.
+
+    Listeners load translations for every loaded component and after config change.
+    """
+
+    async def load_translations(event: Event) -> None:
+        if "language" in event.data:
+            language = hass.config.language
+            _LOGGER.debug(f"Loading translations for language: {language}")
+            await async_load_state_translations_to_cache(hass, language)
+
+    async def load_translations_for_component(event: Event) -> None:
+        component = event.data.get("component")
+        if component is None or "." in str(component):
+            _LOGGER.debug(
+                f"Skipping loading translations for a unsupported component: {component}"
+            )
+            return
+        language = hass.config.language
+        _LOGGER.debug(
+            f"Loading translations for language: {hass.config.language} and component: {component}"
+        )
+        await async_load_state_translations_to_cache(hass, language, [component])
```

Comment:
I still don't think it's useful to support lists here. If needed in the future, we can add it then 👍 
```suggestion
        await async_load_state_translations_to_cache(hass, language, component)
```